### ISSUE TITLE
Marshalling functions do not work with namedtuples

### ISSUE THREAD

**Issue by absassi** [flask-restful/flask-restful#800]
Due to the `isinstance(data, (list, tuple))` test in `marshal` function, types returned by `collections.namedtuple` are iterated as if they were lists, so the following does not work as one would expect:

    import collections
    import json
    from flask_restful import fields, marshal
    schema = {
        "name": fields.String(),
    }
    DataType = collections.namedtuple("DataType", ["name"])
    data = DataType(name="test")
    print(json.dumps(marshal(data, schema)))

A workaround is to convert the namedtuple to dict before marshalling. This works as expected:

    print(json.dumps(marshal(data._asdict(), schema)))

However, sometimes applying the workaround gets complicated when the namedtuples are deeply inside a nested structure of objects.

**Comment by viveksoundrapandi** [comment]
@absassi namedtuples are more of a tuple than a dict . The current implementation it looks perfectly fine if you see this as a list/tuple

**Comment by absassi** [comment]
Yes, they are a tuple and not a dict. However, the schemas don't apply just to dicts, but to objects too, and namedtuples are objects with attributes, like typical ORM classes.

I see no sane use case of `namedtuples` being used as a top-level list container in marshalling, so trying to marshal each element of the namedtuple using the schema makes little sense in this context.
When not in top-level, one would use either `fields.List` for a sequence, or a nested `dict` for a mapping or an object with attributes, so there shouldn't be any ambiguity.

On the other hand, I can see that not considering tuples as list is bad, and specializing for namedtuples may be too much for such a special use case. And, by default, `json.dumps` also serializes namedtuples as tuples...
